### PR TITLE
Optimize GJULE log record source autodetection

### PR DESCRIPTION
**Pull request by avpinchuk**
The `StackWalker` is used to autodetect source class and method.

### CODE REVIEW

File: nucleus/glassfish-jul-extension/src/main/java/org/glassfish/main/jul/record/GlassFishLogRecord.java [hunk 1]
```diff
@@ -289,37 +306,24 @@ public String toString() {
...
-            if (!isIgnoredStackTraceElement(className)) {
-                wrappedRecord.setSourceClassName(className);
-                wrappedRecord.setSourceMethodName(element.getMethodName());
-                return;
-            }
-        }
+        final Optional<StackWalker.StackFrame> stackFrame = STACK_WALKER.walk(stackFrames ->
+            stackFrames.dropWhile(frame -> !isIgnoredStackFrame(frame.getClassName()))
+                .filter(frame -> !isIgnoredStackFrame(frame.getClassName()))
+                .findFirst());
```
Comment (pzygielo) [inline nucleus/glassfish-jul-extension/src/main/java/org/glassfish/main/jul/record/GlassFishLogRecord.java:312]: I'd drop local variable `stackFrame`, continue with `.ifPresent`, and remove import for `Optional`.

**Review by dmatej — APPROVED** [review]

File: nucleus/glassfish-jul-extension/src/main/java/org/glassfish/main/jul/record/GlassFishLogRecord.java [hunk 2]
```diff
@@ -287,41 +289,43 @@ public String toString() {
...
-            || sourceClassName.startsWith("sun.reflect.");
+            "java.lang.Throwable"
+        );
+
+        private static final StackWalker STACK_WALKER = StackWalker.getInstance(StackWalker.Option.SHOW_REFLECT_FRAMES);
+
+        static void detectClassAndMethod(final LogRecord wrappedRecord) {
+            STACK_WALKER
+                .walk(stackFrames ->
+                    stackFrames.dropWhile(frame -> !isIgnoredStackFrame(frame.getClassName()))
```
Comment (pzygielo) [inline nucleus/glassfish-jul-extension/src/main/java/org/glassfish/main/jul/record/GlassFishLogRecord.java:311]: Just for fun - if there only could be `isNotIgnoredStackFrame(StackWalker.StackFrame frame)` method (`::`-referenced in both `dropWhile` and `filter`)...

**Review by arjantijms — APPROVED** [review]

File: nucleus/glassfish-jul-extension/src/main/java/org/glassfish/main/jul/record/GlassFishLogRecord.java (hunk 2 above)
Comment (dmatej) [reply nucleus/glassfish-jul-extension/src/main/java/org/glassfish/main/jul/record/GlassFishLogRecord.java:311]: @avpinchuk @pzygielo Can you do that? It is a good idea, unfortunately @arjantijms already merged the PR, so you have to create new one.

Comment (avpinchuk) [reply nucleus/glassfish-jul-extension/src/main/java/org/glassfish/main/jul/record/GlassFishLogRecord.java:311]: Sure. But as a separate PR ))

**Pull request merged**